astropy/io/misc/pickle_helpers.py

```diff
@@ -43,7 +43,7 @@ def fnunpickle(fileorname, number=0):
 
     """
     if isinstance(fileorname, str):
-        f = open(fileorname, "rb")
+        f = open(fileorname, "rb")  # noqa: PTH123
```

Comment:
These noqas become unneeded if we go with #16972. Moving to draft until it's resolved.